Meeting notes for 2021-03-15 (#147)
solid/authentication-panel: meetings/2021-03-15.md

## Agenda

* [Adding meeting notes from 2021-03-01](https://github.com/solid/authentication-panel/pull/144) -- (15 seconds)
* Possible adjustments to meeting time
* Any discussion/follow-ups from last week’s W3C Credentials WG meeting
*  [Http Sig History missing](https://github.com/solid/authentication-panel/issues/143) (3 minutes) - what to do?
*  [Scope discussion](https://github.com/solid/authentication-panel/issues/86)
* Our use of the webid claim in Solid-OIDC and how we can best support DIDs
* [Clarifying specific areas of overlap with AuthZ #145](https://github.com/solid/authentication-panel/issues/145)

## Minutes

### [Adding meeting notes from 2021-03-01](https://github.com/solid/authentication-panel/pull/144) -- (2 minutes)

Todo - adjust date

### Possible adjustments to meeting time

EU and US times are not aligned for the next two weeks. Let's keep authentication pannel to 10am ET since the only known conflict is 11am ET.

###  [Http Sig History missing](https://github.com/solid/authentication-panel/issues/143) (3 minutes) - what to do?

Git blame shows all contributors. Resolved.

### Any discussion/follow-ups from last week’s W3C Credentials WG meeting

> Aaron: The biggest overlap might be DIDs. If you think about the venn diagram, there is a lot of common interest. Having more interaction

Reviewer: ```suggestion
Aaron: The biggest overlap might be DIDs. If you think about the venn diagram, there is a lot of common interest. Having more interaction ____
```
seems to have been truncated

Dmitri: The confidential storage, then CCG call would be most useful. They are weekly.

* https://github.com/decentralized-identity/confidential-storage/
* https://lists.identity.foundation/g/sds-wg/wiki/18887/94176
* https://lists.identity.foundation/g/sds-wg/wiki

> c

Reviewer: ```suggestion

```

(that's the spec)
github: https://github.com/decentralized-identity/confidential-storage/
this one: https://lists.identity.foundation/g/sds-wg/wiki

###  [Scope discussion](https://github.com/solid/authentication-panel/issues/86)

Aaron: Requesting a specific scope of "solid" seems to be more in line with what OIDC does. It would require an extra burden for the client, but it doesn't seem too cumbersome.

Elf: Would it be sufficient to have a webid claim to match what is in the access token's audience claim?

Aaron: When you request the scope, then the ID token and the access token would include that claim and if you don't, it wouldn't be there. It's what happens with email claims for example.

Dmitri: I think requiring a scope is a good idea, it sounds like a decent way to work with the mechanism.

Aaron: Can we get feedback/concerns about client tooling for example?

Henry: we could put a note in the spec linking to the issue so that people can be aware of the question we are asking.

Matthieu: Spec level changelog? Would it be good to have those for all the Solid specs

Aaron: I've seen it in other specs and it is useful.

> Sarven: It's specifically useful if you have significant snapshots of the spec (WD, CR, PR..)

Reviewer: ```suggestion
Sarven: It's specifically useful if you have significant snapshots of the spec (WD, CR, PR, ...)
```

Henry: Could we use the git history?

Aaron: There's a lot of info in history, it will be there among people fixing less significant things.

### Our use of the webid claim in Solid-OIDC and how we can best support DIDs

Aaron: How to best support DIDs? Right now we have a webid claim and we'd like to support dids. Even though the WebID says it must be an https URL, we sort of signal it could be a DID. We could be more specific and rename the claim to solid. So that would be future proof for supporting VCs, DIDs...

Dmitri: +1 to Aaron's suggestion to have a general `solid` claim that specifies both WebID and DID.

Matthieu: +1 to Aaron's suggestion.

Justin: That makes a lot of sense.

Aaron: I spoke with Ruben V. last week and he was very supportive, he suggested making the value space for this claim IRIs.

> Henry: Which DID do you support then? You could keep saying WebIDs because people argument they are more than HTTPS. That's a limitation we allowed to be formulated to finalize the spec (DID took another 7 years to mature).

Reviewer: ```suggestion
Henry: Which DID do you support then? You could keep saying WebIDs because people argument they are more than HTTPS. That's a limitation we allowed to be formulated to finalize the spec (DID took another 7 years to mature).
```
It should perhaps be noted that WebID spec has never been finalized; it remains a DRAFT document, and has never gone through full RecTrack process.

Author: It would be like saying that foaf:Person needs to go through full W3C standard track. WebID just is a URI with a particular referent following RDF rules.

Author: Actually I need to look at this write up of what I said, as it seems a bit mangled.

# 2021-03-15 Authentication Panel

## Meeting
 * Call: https://meet.jit.si/solid-authentication

## Present

* Henry Story
* Aaron Coburn
* Dmitri Zagidulin
* Matthieu Bosquet
* Sarven Capadisli
* Justin Bingham

## Agenda

* [Adding meeting notes from 2021-03-01](https://github.com/solid/authentication-panel/pull/144) -- (15 seconds)
* Possible adjustments to meeting time
* Any discussion/follow-ups from last week’s W3C Credentials WG meeting
*  [Http Sig History missing](https://github.com/solid/authentication-panel/issues/143) (3 minutes) - what to do?
*  [Scope discussion](https://github.com/solid/authentication-panel/issues/86)
* Our use of the webid claim in Solid-OIDC and how we can best support DIDs
* [Clarifying specific areas of overlap with AuthZ #145](https://github.com/solid/authentication-panel/issues/145)

## Minutes

### [Adding meeting notes from 2021-03-01](https://github.com/solid/authentication-panel/pull/144) -- (2 minutes)

Todo - adjust date

### Possible adjustments to meeting time

> EU and US times are not aligned for the next two weeks. Let's keep authentication pannel to 10am ET since the only known conflict is 11am ET.

Reviewer: ```suggestion
EU and US times are not aligned for the next two weeks. Let's keep authentication panel to 10am ET since the only known conflict is 11am ET.
```

Henry: we could put a note in the spec linking to the issue so that people can be aware of the question we are asking.

Matthieu: Spec level changelog? Would it be good to have those for all the Solid specs

Aaron: I've seen it in other specs and it is useful.

Sarven: It's specifically useful if you have significant snapshots of the spec (WD, CR, PR, ...)

Henry: Could we use the git history?

Aaron: There's a lot of info in history, it will be there among people fixing less significant things.

### Our use of the webid claim in Solid-OIDC and how we can best support DIDs

Aaron: How to best support DIDs? Right now we have a webid claim and we'd like to support dids. Even though the WebID says it must be an https URL, we sort of signal it could be a DID. We could be more specific and rename the claim to solid. So that would be future proof for supporting VCs, DIDs...

Dmitri: +1 to Aaron's suggestion to have a general `solid` claim that specifies both WebID and DID.

Matthieu: +1 to Aaron's suggestion.

Justin: That makes a lot of sense.

Aaron: I spoke with Ruben V. last week and he was very supportive, he suggested making the value space for this claim IRIs.

> Henry: The problem remains which DID should we support? This was the question Ruben asked initially on the [Support DID issue](https://github.com/solid/specification/issues/217#issuecomment-532521655). One could argue that WebIDs could still be extended to cover DIDs. Indeed the WebID group considered for a long time covering more than `https` URLs. But we ended up adding the limitation in order to help finalize the spec. After all: DID took another 7 years to mature.

Reviewer: Again (still), the [WebID spec](https://www.w3.org/2005/Incubator/webid/spec/identity/) is NOT finalized. It remains an Editor's Draft, and says (multiple times!) in the Status of this Document --

> This is a draft document and may be updated, replaced or obsoleted by other documents at any time. It is inappropriate to cite this document as other than work in progress.

It is extremely problematic to treat this DRAFT document as a "finalized spec".

Author: Manu Sporny mentioned in the presentation last week and in [another one just before that](https://twitter.com/bblfish/status/1368897757365288965) just how problematic it was to get a lot of this stuff accepted at the W3C until recently. (I think he was speaking of identity in particular). So I am happy I did not spend all these years fighting to get this through and spent time on other things. Perhaps now or soon will be the right time to finish it off and present it. Many specs in everywhere take many years to incubate and can even gain wide adoption before they get blessed.

> Henry: The problem remains which DID should we support? This was the question Ruben asked initially on the [Support DID issue](https://github.com/solid/specification/issues/217#issuecomment-532521655). One could argue that WebIDs could still be extended to cover DIDs. Indeed the WebID group considered for a long time covering more than `https` URLs. But we ended up adding the limitation in order to help finalize the spec. After all: DID took another 7 years to mature.

Reviewer: Possibly relevant comments from @msporny notwithstanding, my point remains: the WebID spec should not be referred to as anything like _finalized_ because it's not! Its current shape was not even officially published as an Incubation Group (XG) Note (roughly comparable to what today would be a Community Group [CG] Report)!

Further, having been there at the time, the restriction therein to `https` URLs resulted from a flawed poll (it was issued by a single person, not the group as a whole; it was emitted through a system that bestowed improper gravity by suggesting its origination was not a single person; wording of several questions was ill considered; etc.), which was treated by some as a binding vote, treated by others as a temperature-taking straw poll, and entirely ignored by others.

Therefore --

```suggestion
Henry: The problem remains which DID should we support? This was the question Ruben asked initially on the [Support DID issue](https://github.com/solid/specification/issues/217#issuecomment-532521655). One could argue that WebIDs could still be extended to cover DIDs. Indeed the WebID group considered for a long time covering more than `https` URLs. But we ended up adding the limitation in order to help move the [WebID XG draft spec](https://www.w3.org/2005/Incubator/webid/spec/identity/) forward. After all: DID took another 7 years to mature.
```

Justin: If we were to take the proposal, what's the worry

Elf: Client support only https the client give webid, gets it and then comes the response. If the client got a did:web, it can't get the info it needs. We need to ensure there's a response with a supported protocol for all parties. Maybe it's not a problem but we should consider and think it through.

Henry: I think the same problem applies to HTTPSig I've left it open for key IDs but for credentials I would leave it open. It's kind of content negotiation on URIs.

Matthieu: Are there any specifics at the authentication protocol level for the different kinds of DIDs? Will they all be compatible with Solid-OIDC?

Dmitri: The DID community is specialising on the https://identity.foundation/did-siop/

...: They're trying to centralise on DID authentication protocols. The couple of protocol that there are involve a wallet based authentication and the self issued OpenID connect.

Sarven: Wait for maturity for requiring did:web on the Solid protocol level.

Dmitri on Gitter: DIDs have been in W3C WG for about a year now, and they're going to CR either this week, or this past week.

> 

Reviewer 2: ```suggestion

Sarven on Gitter: we are not talkin gabout DID in generall.. but did:web specifically at the moment

Dmitri on Gitter: +1 agree with Sarven. tho obviously I'm a fan of DIDs, the solid community needs to get more experience with em etc

```

Sarven: Wait for maturity for requiring did:web on the Solid protocol level.

Dmitri on Gitter: DIDs have been in W3C WG for about a year now, and they're going to CR either this week, or this past week.

Justin: What constitutes maturity? If we were to compare DIDs, for example did:web and WebIDs, would there be a big maturity gap?

Sarven: There's a reason why the Solid ecosystem started with WebIDs, replacing WebIDs with DIDs would be premature. They can be compatible and extended.

Henry: There is a reason for WebIDs: HTTP GET PUT POST DELETE using Solid protocol is really mature. You have servers for HTTPS, all you need is a GET and an RDF library. DIDs are more experimental. We should analyse per case.

Aaron: 1 minute left.

...: Long detour on DID protocols. The initial suggestion is not which protocol to support but do we need the door open to extend support for any kind of such identifiers in the future? Just so that how would one be able to implement this. Likewise with VCs. By boxing ourselves with webid, it's pretty narrow, by using "solid", we're very open to the future.

> From Gitter:

Reviewer 3: ```suggestion
From Gitter: https://gitter.im/solid/authentication-panel?at=604f75c9823b6654d2aa04a1
```
We could add the link.

Dmitri: +1 agree with Sarven. tho obviously I'm a fan of DIDs, the solid community needs to get more experience with em etc
justinweb: so i think aaron’s proposal leaves the door open without committing +1 for solid claim, +1 for IRI
bblfish (henry): +1 for leaving the door open, but if we leave the door open, we should specify how extensions are going to work.
matthieu: +1 to leave the door open, +1 for solid claim, +1 fo IRI
elf: +1 to keep it open, preferably with explicit discovery / negotiation

csarven: Proposed Action: Rename to solid and IRI range. Solid OIDC editors to document interop re diff IRIs

> Gitter: +1s https://gitter.im/solid/authentication-panel?at=604f769122a5ce4a91599df9
> 

Reviewer 3: ```suggestion
```
This is now transcribed above.
